proposed/2022/package-status-indicators.md

```diff
@@ -0,0 +1,120 @@
+
+Building upon Tim's initial mockup, there should be a UI affordance that demonstrates the truth of each package's status in the solution explorer representing these three words.
+
+![](../../meta/resources/PackageStatusIndicators/package-status-indicators.png)
+
+For the sake of not annoying developers, we should only bubble up a warning affordance to the top-level node (Packages and Dependencies) if there is a known top-level vulnerability or deprecated package in their package graph.
+
+Outdated packages would still show the indicators if the packages node is expanded however.
+
+Currently, NuGet only uses two icons to display status indicators:
+
+- A warning triangle (vulnerability and deprecation).
+![](../../meta/resources/PackageStatusIndicators/StatusWarning.png)
+- An up arrow (outdated).
+![](../../meta/resources/PackageStatusIndicators/StatusUpdateAvailable.png)
+
+Hovering over the package and icon will display the respective information for the status.
+
+- Deprecated - "This package has been deprecated as it is `<deprecation category>`."
+- Vulnerable - "This package has at least one vulnerability with `<severity category>` severity."
+- Outdated - "This package has version `<latest version>` available.
+
+When right clicking a package, a developer should be able to get to the "Manager NuGet Packages..." context menu action like they can with the "Packages" and "Dependencies" nodes.
+
+Because this feature may need some experimentation to get the right signal from the noise and the potential for exponential transitive-level vulnerabilities, developers will need a way to opt-out from the experience as a whole, and options to opt-in to varying levels of information.
+
+This may need to include a new options panel inside the `Projects and Solutions > SDK-Style Projects` regarding the Package Settings where there's a top-level checkbox for the feature, and either a drop-down indicating a package status level or multiple checkboxes for each status mentioned above.
+
+<!-- ### Technical explanation -->
+
+<!-- Explain the proposal in sufficient detail with implementation details, interaction models, and clarification of corner cases. -->
+
+## Drawbacks
+
+<!-- Why should we not do this? -->
+While developers continue to seek this information in more locations, it can also be quite a bit of noise towards the general development experience. Developers tend to compare visual issues in the solution explorer akin to ["DLL Hell"](https://en.wikipedia.org/wiki/DLL_Hell) which we would like to avoid the perception as much as possible while providing valuable information about dependencies.
+
+This feature can potentially collide with existing experiences in which package dependencies may not exist on disk and/or assets file are incorrect and require a restore to resolve. There can also be false positives/negatives with the current state of the project system and the need to reload projects to refresh the package status in realtime.
```

Comment:
Could you explain more about false positives that would require reloading projects? I'm not aware of such a situation, for projects opened by the .NET Project System at least.